src/Snt/Capi/Http/ApiHttpClientConfiguration.php

```diff
@@ -39,6 +40,23 @@ public function __construct($endpoint, $apiKey, $apiSecret)
     }
 
     /**
+     * @param array $config
+     *
+     * @return ApiHttpClientConfiguration
+     * @throws InvalidArgumentException
+     */
+    public static function fromArray(array $config)
+    {
+        foreach (['endpoint', 'apiKey', 'apiSecret'] as $key) {
+            if (empty($config[$key])) {
+                throw new InvalidArgumentException(sprintf('Missing config parameter "%s"', $key));
```

Comment:
even with `MissingApiHttpConfigurationBecauseOfEmptyArrayIndexException`. This exception is thrown when at least one of `$config` array element is empty or not set. The only way to make it more descriptive is to introduce separate exception types for each `$config` key. Anyway - IMHO during development exception messages are more descriptive than types. `InvalidArgumentException` fits here, we can use it because it has no sense to separate SDK from PHP exceptions types.